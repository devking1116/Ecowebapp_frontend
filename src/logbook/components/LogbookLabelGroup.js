/* @flow */
import React from 'react';
import reactCSS from 'reactcss';

type Props = {
  showAll?: bool,
  children?: any,
};

/**
 * A component that displays Labels marking logbooks.
 */
export default function LogbookLabelGroup({ children, showAll, ...otherProps }: Props): React.Element<*> {
  const [ topLabel, ...otherLabels ] = React.Children.toArray(children);

  const styles = reactCSS({
    'default': {
      wrapper: {
        display: 'flex',
        marginTop: 5,
      },
      label: {
        position: 'relative',
        zIndex: otherLabels.length + 1,
        marginLeft: 5,
      },
    }
  });

  return (
    <div style={styles.wrapper} {...otherProps}>
      <div style={styles.label}>
        {topLabel}
      </div>
    {otherLabels.map((currLabel, index) =>
      <div
        style={{
          ...styles.label,
          left: showAll? undefined: -50 * (index + 1),
          zIndex: styles.label.zIndex - (index + 1),
        }}
        key={index}
      >
        {currLabel}
      </div>
    )}
    </div>
  );
}
