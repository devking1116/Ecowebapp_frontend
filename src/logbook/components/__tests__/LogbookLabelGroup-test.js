import React from 'react';
import renderer from 'react-test-renderer';

import LogbookLabelGroup from '../LogbookLabelGroup';

test('it displays a single label when only one is given.', () => {
  const component = renderer.create(
    <LogbookLabelGroup>
      <div>Test Label</div>
    </LogbookLabelGroup>
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('it displays the entier top label on top of the remaining ones when multiple are given.', () => {
  const component = renderer.create(
    <LogbookLabelGroup>
      <div>Test Label 0</div>
      <div>Test Label 1</div>
      <div>Test Label 2</div>
    </LogbookLabelGroup>
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('it displays the entier element of every label when multiple labels and the showAll prop are given.', () => {
  const component = renderer.create(
    <LogbookLabelGroup showAll>
      <div>Test Label 0</div>
      <div>Test Label 1</div>
      <div>Test Label 2</div>
    </LogbookLabelGroup>
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
