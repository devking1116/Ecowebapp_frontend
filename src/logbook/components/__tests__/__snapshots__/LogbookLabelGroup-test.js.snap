exports[`test it displays a single label when only one is given. 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "marginTop": 5,
    }
  }>
  <div
    style={
      Object {
        "marginLeft": 5,
        "position": "relative",
        "zIndex": 1,
      }
    }>
    <div>
      Test Label
    </div>
  </div>
</div>
`;

exports[`test it displays the entier element of every label when multiple labels and the showAll prop are given. 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "marginTop": 5,
    }
  }>
  <div
    style={
      Object {
        "marginLeft": 5,
        "position": "relative",
        "zIndex": 3,
      }
    }>
    <div>
      Test Label 0
    </div>
  </div>
  <div
    style={
      Object {
        "left": undefined,
        "marginLeft": 5,
        "position": "relative",
        "zIndex": 2,
      }
    }>
    <div>
      Test Label 1
    </div>
  </div>
  <div
    style={
      Object {
        "left": undefined,
        "marginLeft": 5,
        "position": "relative",
        "zIndex": 1,
      }
    }>
    <div>
      Test Label 2
    </div>
  </div>
</div>
`;

exports[`test it displays the entier top label on top of the remaining ones when multiple are given. 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "marginTop": 5,
    }
  }>
  <div
    style={
      Object {
        "marginLeft": 5,
        "position": "relative",
        "zIndex": 3,
      }
    }>
    <div>
      Test Label 0
    </div>
  </div>
  <div
    style={
      Object {
        "left": -50,
        "marginLeft": 5,
        "position": "relative",
        "zIndex": 2,
      }
    }>
    <div>
      Test Label 1
    </div>
  </div>
  <div
    style={
      Object {
        "left": -100,
        "marginLeft": 5,
        "position": "relative",
        "zIndex": 1,
      }
    }>
    <div>
      Test Label 2
    </div>
  </div>
</div>
`;
